pkg/controller/keyvault/kv_util.go

```diff
@@ -0,0 +1,115 @@
+package keyvault
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/Azure/aks-app-routing-operator/api/v1alpha1"
+	"github.com/Azure/aks-app-routing-operator/pkg/config"
+	kvcsi "github.com/Azure/secrets-store-csi-driver-provider-azure/pkg/provider/types"
+	v1 "k8s.io/api/networking/v1"
+	"net/url"
+	secv1 "sigs.k8s.io/secrets-store-csi-driver/apis/v1"
+	"strings"
+)
+
+var NginxNamePrefix = "keyvault-nginx-"
+
+func BuildSPC(ic interface{}, spc *secv1.SecretProviderClass, config *config.Config) (bool, error) {
+	var certURI, specSecretName string
+
+	if nic, ok := ic.(*v1alpha1.NginxIngressController); ok {
+		if nic.Spec.IngressClassName == "" {
+			return false, nil
+		}
+		if nic.Spec.DefaultSSLCertificate == nil || nic.Spec.DefaultSSLCertificate.KeyVaultURI == nil {
+			return false, nil
+		}
+		certURI = *nic.Spec.DefaultSSLCertificate.KeyVaultURI
+		specSecretName = DefaultNginxCertName(nic)
+	}
+
+	if ing, ok := ic.(*v1.Ingress); ok {
```

Comment:
the type check should probably be a switch statement with a default of returning an error due to incorrect ic type

Reply:
more robust and catches type errors